src/App.tsx

```diff
@@ -1,51 +1,88 @@
+import classNames from 'classnames';
 import React from 'react';
 import './App.scss';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+type State = {
+  selected: string[];
 };
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selected: ['Jam'],
+  };
+
+  render() {
+    const { selected } = this.state;
+
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
+
+            {selected.length
+              ? `${selected} ${selected.length > 1 ? 'are' : 'is'} selected`
+              : 'No goods selected'}
+          </h1>
+
+          {selected.length > 0 && (
+            <button
+              type="button"
+              className="App__clear button"
+              onClick={() => {
+                this.setState({ selected: [] });
+              }}
+            >
+              Clear
+            </button>
+          )}
+        </header>
+
+        <ul className="App__list">
+          {goodsFromServer.map(good => (
+            <li
+              className={classNames(
+                'Good',
+                { 'Good--active': selected.includes(good) },
+              )}
+              key={good}
+            >
+              {good}
+
+              {selected.includes(good)
+                ? (
+                  <button
+                    type="button"
+                    className="Good__remove button"
+                    key={good}
+                    onClick={() => {
+                      selected.splice(selected
+                        .indexOf(good), 1);
+                      this.setState({ selected });
+                    }}
```

Comment:
Splice is a mutable method, don't use splice. U can't mutate state, let's use filter and move to a separate method